client/extensions/woocommerce/state/sites/http-request.js

```diff
@@ -30,6 +31,20 @@ const _request = ( method, path, siteId, body, action ) => {
 };
 
 /**
+ * Prepares a request action that will return the body and headers.
+ * The resulting data will be in the form of `{ data: { body: { API data }, headers: { API response headers } } }`
```

Comment:
It's worth mentioning here that the real status code of a request will be returned here as well, and that the HTTP status that the data layer gets will always be 200. See the example here: https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_envelope

Reply:
Thanks for the pointer.  I'll add some code to check that.

Reply:
Added in 4dcf8a0